tdh/resource_data_plane.go

```diff
@@ -94,55 +96,68 @@ func (r *dataPlaneResource) Schema(ctx context.Context, _ resource.SchemaRequest
 				Optional:    true,
 			},
 			"auto_upgrade": schema.BoolAttribute{
-				Description: "Whether to enable auto-upgrade on Data plane",
-				Required:    true,
+				MarkdownDescription: "Whether to enable auto-upgrade on Data plane.\n Note: This field should be set to false for TAS data-plane creation ",
+				Optional:            true,
+				Required:            false,
 			},
 			"cp_bootstrapped_cluster": schema.BoolAttribute{
-				Description: "Whether to onboard Data Plane on a K8s cluster running TDH Control Plane",
-				Optional:    true,
+				MarkdownDescription: "Whether to onboard Data Plane on a K8s cluster running TDH Control Plane \n Note: Not a required field during TAS data-plane creation",
+				Optional:            true,
 			},
 			"org_id": schema.StringAttribute{
-				Description: "Organization ID",
+				Description: "Organization ID. This filed is not required during TAS data-plane creation",
 				Optional:    true,
 			},
 			"provider_name": schema.StringAttribute{
 				Description: "Provider name",
-				Computed:    true,
+				Required:    true,
 			},
 			"backup_storage_class": schema.StringAttribute{
-				Description: "Backup Storage Class will be used to create all backups on this data plane. Please note this cannot be changed in future.",
-				Required:    true,
+				MarkdownDescription: "Backup Storage Class will be used to create all backups on this data plane. Please note this cannot be changed in future.\n Note: This field is non-mandatory during the TAS data plane creation. \n It is a mandatory field during  Non TAS (i.e `tkgm`, `tkgs`, `openshift`) data plane creation.",
+				Required:            false,
+				Optional:            true,
 			},
 			"data_plane_release_id": schema.StringAttribute{
-				MarkdownDescription: "ID of the Helm Release. Please use datasource `tdh_data_plane_helm_releases` to get this.",
-				Required:            true,
+				MarkdownDescription: "ID of the Helm Release. Please use datasource `tdh_data_plane_helm_releases` to get this.\n Note: This field is non-mandatory during the TAS data plane creation. \n It is a mandatory field during  Non TAS (i.e `tkgm`, `tkgs`, `openshift`) data plane creation.",
+				Required:            false,
+				Optional:            true,
 			},
 			"storage_classes": schema.SetAttribute{
-				Description: "Storage Classes on the data plane",
-				ElementType: types.StringType,
-				Required:    true,
+				MarkdownDescription: "Storage Classes on the data plane. \n Note: This field is non-mandatory during the TAS data plane creation . \n It is a mandatory field during Non TAS (i.e `tkgm`, `tkgs`, `openshift`)	 data plane creation.",
+				ElementType:         types.StringType,
```

Comment:
here also use proper format, else it looks normal on registry documentation. Also while fixing, go to registry datasource/resource documentation, and see if anything is looking like unexpected, or if from user's POV, will it be sufficient or helpful to create resource or fetch a datasource (this part u can do in separate PR)